subsys/bluetooth/host/hci_core.c

```diff
@@ -3641,6 +3641,46 @@ int bt_enable(bt_ready_cb_t cb)
 	return 0;
 }
 
+int bt_disable(void)
+{
+	int err;
+
+	if (!bt_dev.drv) {
+		BT_ERR("No HCI driver registered");
+		return -ENODEV;
+	}
+
+	atomic_clear_bit(bt_dev.flags, BT_DEV_ENABLE);
+
+	if (atomic_test_and_set_bit(bt_dev.flags, BT_DEV_DISABLE)) {
+		return -EALREADY;
+	}
+
+	err = bt_dev.drv->close();
```

Comment:
You can't assume that drivers implement a `close()` callback. This is important for backwards compatibility, and also if a `close()` callback doesn't make sense for some drivers (possibly most of them). So you should probably have a check right after the `if (!bt_dev.drv)` check:
```c
if (!bt_dev.drv->close) {
	return -ENOTSUP;
}
```

Reply:
Nice, sure. Added.